### PR TITLE
community: fix a variety of bugs in Louvain and Q

### PR DESCRIPTION
**Pull request by kortschak**
I must remember to always include a sparsely ID'd node set test.

@vladimir-ch Please take a look.

### CODE REVIEW

**Comment by kortschak** [comment]
Wait. It looks like there are more.

**Comment by kortschak** [comment]
The last commit here is a prelude to implementing Smart Local Moving which repeats the application of the local moving heuristic on each subnetwork.

File: undirect.go [hunk 1]
```diff
@@ -125,7 +125,7 @@ func (g Undirect) Weight(x, y Node) (w float64, ok bool) {
 	re := g.G.Edge(y, x)
 
 	var f, r float64
-	if wg, ok := g.G.(Weighter); ok {
+	if wg, wOk := g.G.(Weighter); wOk {
```
Comment (vladimir-ch) [inline undirect.go]: Now I remember that I saw this during the review but then somehow forgot to comment. Sorry.

**Comment by vladimir-ch** [comment]
LGTM

**Pull request merged**